Update t0-118 topo file configurations

dayouliu1 opened the pull request:
Added an additional uplink and vlan, as well as fix existing config bugs and consistency.

<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/sonic-net/SONiC/blob/gh-pages/CONTRIBUTING.md

Please provide following information to help code review process a bit easier:
-->
### Description of PR
<!--
- Please include a summary of the change and which issue is fixed.
- Please also include relevant motivation and context. Where should reviewer start? background context?
- List any dependencies that are required for this change.
-->

Summary:
Fixes # (issue)

Update Arista-7260CX3-D108C10 t0-118 topo file to include an additional vlan config for the first two interfaces (both 100G).

Additionally updated config errors and consistency. 

### Type of change

<!--
- Fill x for your type of change.
- e.g.
- [x] Bug fix
-->

- [x] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] New Test case
    - [ ] Skipped for non-supported platforms
- [ ] Test case improvement


### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [ ] 202405
- [x] 202411

### Approach
#### What is the motivation for this PR?
Update t0 118 topo yaml file.

#### How did you do it?
Added one additional vlan config for interfaces 0, 1
Changed two_vlan_a config consistency
Fixed missing interfaces

#### How did you verify/test it?
Tested add_topo on a t0-118 setup 

#### Any platform specific information?

#### Supported testbed topology if it's a new test case?

### Documentation
<!--
(If it's a new feature, new test case)
Did you update documentation/Wiki relevant to your implementation?
Link to the wiki page?
-->

mssonicbld commented:
/azp run

mssonicbld commented:
/azp run

ZhaohuiS commented on ansible/vars/topo_t0-118.yml:
```diff
@@ -1,7 +1,5 @@
 topology:
   host_interfaces:
-   - 0
```
@dayouliu1 why remove 0 and 1 from interfaces?
Arista-7260CX3-D108C8: 

Interfaces: 108*50G + 8*100G 

Port_config.ini 

[sonic-buildimage/device/arista/x86_64-arista_7260cx3_64/Arista-7260CX3-D108C8/port_config.ini at 905c666e02e01c2508f2beb1a195333338f4c7fb · sonic-net/sonic-buildimage](https://github.com/sonic-net/sonic-buildimage/blob/905c666e02e01c2508f2beb1a195333338f4c7fb/device/arista/x86_64-arista_7260cx3_64/Arista-7260CX3-D108C8/port_config.ini) 

Topology: T0-116 

Arista-7260CX3-D108C10 

Interfaces: 108*50G + 10*100G 

Port_config.ini 

[sonic-buildimage/device/arista/x86_64-arista_7260cx3_64/Arista-7260CX3-D108C10/port_config.ini at 905c666e02e01c2508f2beb1a195333338f4c7fb · sonic-net/sonic-buildimage](https://github.com/sonic-net/sonic-buildimage/blob/905c666e02e01c2508f2beb1a195333338f4c7fb/device/arista/x86_64-arista_7260cx3_64/Arista-7260CX3-D108C10/port_config.ini) 

Topology:T0-118
The left is D108C10, the right one is D108C08.

<img width="881" alt="image" src="https://github.com/user-attachments/assets/7607495e-7ab8-4525-b67a-51b85879e572" />

ZhaohuiS replied:
@dayouliu1 can you give an explanation why change like this?

ZhaohuiS commented on ansible/vars/topo_t0-118.yml:
```diff
@@ -113,51 +111,56 @@ topology:
         Vlan1000:
           id: 1000
-          intfs: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62]
-          prefix: 192.168.0.1/25
```
@dayouliu1  The IPv4/IPv6 range is set it like this on purpose, otherwise, the testcases in test_acl will fail.
https://github.com/sonic-net/sonic-mgmt/blob/master/tests/acl/test_acl.py

```
# Below M0_VLAN IPs are ip in vlan range
# Note: If two_vlan_a is used, the IP range should choose
# DOWNSTREAM_DST_IP_VLAN and DOWNSTREAM_DST_IP_VLAN2000
# separately for each vlan
DOWNSTREAM_DST_IP_VLAN = {
    "ipv4": "192.168.0.123",
    "ipv6": "fc02:1000::5"
}
DOWNSTREAM_IP_TO_ALLOW_VLAN = {
    "ipv4": "192.168.0.122",
    "ipv6": "fc02:1000::6"
}
DOWNSTREAM_IP_TO_BLOCK_VLAN = {
    "ipv4": "192.168.0.121",
    "ipv6": "fc02:1000::7"
}

DOWNSTREAM_DST_IP_VLAN2000 = {
    "ipv4": "192.168.0.253",
    "ipv6": "fc02:1000:0:1::5"
}
DOWNSTREAM_IP_TO_ALLOW_VLAN2000 = {
    "ipv4": "192.168.0.252",
    "ipv6": "fc02:1000:0:1::6"
}
DOWNSTREAM_IP_TO_BLOCK_VLAN2000 = {
    "ipv4": "192.168.0.251",
    "ipv6": "fc02:1000:0:1::7"
}
```

dayouliu1 commented:
@ZhaohuiS

ZhaohuiS commented:
@dayouliu1 didn't see your reply for my question? 
why remove 0 and 1 from interfaces? I removed 22, 23 from host_interfaces and add 30,31 back as comparing top_t0-116. Any mistake for my change? Could you please clarify the reason why you make change like this?

dayouliu1 commented:
> @dayouliu1 didn't see your reply for my question? why remove 0 and 1 from interfaces? I removed 22, 23 from host_interfaces and add 30,31 back as comparing top_t0-116. Any mistake for my change? Could you please clarify the reason why you make change like this?

Yes so just to clarify my responses above. In Arista-7260CX3-D108C10/port_config.ini we see interface 0 and 1 is configured to 100G speeds and use all 4 lanes:
```
Ethernet0       77,78,79,80       Ethernet1/1  1      100000
Ethernet4       65,66,67,68       Ethernet2/1  2      100000
```

I am not saying there is a mistake, I am just bringing up for discussion if we want to consider using interface 0 and 1 for uplinks as well since the other 100G interfaces are used for uplinks (Ex: Ethernet48 - Ethernet76). Hence why I removed interface 0 and 1 from host interfaces (as it will be used for uplinks).

For the t0-116 configuration, in Arista-7260CX3-D108C8/port_config.ini, we see that interface 0 and 1 is only 50G, so it is not skipped in the host_interaces config:
```
Ethernet0       77,78             Ethernet1/1  1      50000
Ethernet2       79,80             Ethernet1/3  1      50000
```

ZhaohuiS commented:
@dayouliu1  @StormLiangMS 
From what I found, prod device still use these 2 100G ports connect to Server node, they are used as downlink, not uplink.

```
hostname#show interface status
Port       Name                              Status       Vlan     Duplex Speed  Type         Flags Encapsulation
Et1/1      NOT CONNECTED                     disabled     routed   full   100G   Not Present                     
Et1/3                                        inactive     routed   full   50G    Not Present                     
Et2/1      SAT111121507002:NIC1.PORT1        connected    807      a-full a-100G 100GBASE-CR4                    
Et2/3                                        inactive     routed   full   50G    100GBASE-CR4  
```
